analysis/quantifier-comparisons/01-import-quant-data.Rmd

```diff
@@ -147,6 +232,235 @@ alevin_sces <- alevin_quant_ids  %>%
 names(alevin_sces) <- alevin_quant_ids
 ```
 
+
+```{r}
+# for every combination of mrna + intronic region create a collapsed counts matrix that contains the sum of the total counts for that gene (mrna + intron)
+# we could use tximeta::splitSE but it just creates two separate matrices rather than one combined one 
+collapse_intron_counts <- function(counts) {
+  # find the shared genes in that counts matrix 
+  shared_genes = intersect(row.names(counts), rownames(intron_metadata))
+  # replace row names with -I appended with corresponding spliced gene 
+  row.names(counts)[which(row.names(counts) %in% shared_genes)] = intron_metadata[shared_genes, "spliced"]
+  # aggregate Matrix counts by gene name 
+  aggregate.Matrix(counts, row.names(counts))
+}
+
+```
+
+
+```{r}
+# collapse intron counts for only counts matrices containing intron regions
+alevin_nucleus_sces <- alevin_sces[grep("spliced_intron", names(alevin_sces))] %>%
+  purrr::map(
+    ~ counts(.x)
+  ) %>%
+  purrr::map(
+    ~ collapse_intron_counts(.x)
+  ) %>%
```

Comment:
Since these are one argument functions, we can simplify, I think (also incorporating the `intron_metadata` argument from above): 

I'll also note we do these three steps a few times, so a function to do it all seems reasonable.

```suggestion
  purrr::map(counts) %>%
  purrr::map(collapse_intron_counts, intron_metadata) %>%
```